### PR TITLE
Publish confirmation and other post alert changes.

### LINKED ISSUE #6397: Posting: Make it harder to accidentially publish an incomplete post
A user has shared their frustration regarding accidentally published posts and asks if there could be an "Are you sure?" prompt to help prevent premature publishing.

What could we do to help avoid accidentally publishing an incomplete post?

### PR DESCRIPTION
**Pull request by diegoreymendez**
### Description:

Improves the publishing experience by adding a confirmation alert.

Also renames the cancel button within "More".  It's now named "Keep Editing".

Includes some minor code cleanup.

Fixes #6397 

### Screenshots:

- **Publish confirmation alert**:

screen shot 2018-03-05 at 13 53 16

- **More button alert**:

screen shot 2018-03-05 at 13 53 29

### Testing:

**Test 1:**

- Create a new post.
- Write a title and some content.
- Tap publish and make sure there's a confirmation alert as described.

**Test 2:**

- Create a new post.
- Tap X
- When the alert comes up, pick "Save as Draft".
- Make sure the publish confirmation alert does not come up.

**Test 3:**

- Create a new post.
- Tap on "..."
- Make sure the default action is now named "Keep Editing".

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Aztec/ViewControllers/AztecPostViewController.swift [hunk 1]
```diff
@@ -2542,8 +2538,52 @@ private extension AztecPostViewController {
...
+            if dismissWhenDone {
+                self.dismissOrPopView(didSave: true)
+            } else {
+                self.createRevisionOfPost()
+            }
+        }
+    }
 
-    func publishPost(completion: ((_ post: AbstractPost?, _ error: Error?) -> Void)? = nil) {
+    /// Published the post
```
Comment (frosty) [inline WordPress/Classes/ViewRelated/Aztec/ViewControllers/AztecPostViewController.swift]: Typo: `Published` -> `Publish`

**Comment by iamthomasbishop** [comment]
Agreed w/ @frosty - let's make it an alert on iPad. If we're doing that, would it make sense to make it an alert in both cases, for consistency?

**Comment by diegoreymendez** [comment]
@frosty - Good catches.  The fix is up.

@iamthomasbishop - As discussed via private chat, we're keeping the action sheet for iPhone, and showing an alert for iPad.

**Comment by diegoreymendez** [comment]
I'd be ok with switching if that's the direction we're taking.  @iamthomasbishop - thoughts?

@iamthomasbishop - Side note: sometimes when working on UI stuff it feels to me like we could have a UI guide of some kind, explaining what to use in each scenario.  Not a big deal, but something that could be worked on the side.

**Comment by diegoreymendez** [comment]
As discussed via DM, we're sticking with the standard alert for now.

**Pull request merged**